src/test-utils/with-app-test-wrapper.js

```diff
@@ -0,0 +1,27 @@
+import { DataProvider } from '@dhis2/app-runtime'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { MemoryRouter as Router, Route } from 'react-router-dom'
+import { QueryParamProvider } from 'use-query-params'
+import { CurrentItemProvider } from '../shared/index.js'
+import QueryClientWrapper from './query-client-wrapper.js'
+
+export const withAppTestWrapper = (
```

Comment:
I think a better strategy could be to have a custom render function: https://testing-library.com/docs/react-testing-library/setup/#custom-render.

If we need a different initial url, I'd propose to expose the individual building blocks used by the custom render function, e.g. `withAppTestWrapper` and `originalRender`, from the `test-utils.js` module as well